capstone/src/main/java/com/google/sps/servlets/ConnectionPoolContextListener.java

```diff
@@ -0,0 +1,176 @@
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.sps.servlets;
+
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.servlet.annotation.WebListener;
+import javax.sql.DataSource;
+
+@WebListener("Creates a connection pool that is stored in the Servlet's context for later use.")
+public class ConnectionPoolContextListener implements ServletContextListener {
+
+    // Saving credentials in environment variables is convenient, but not secure - consider a more
+    // secure solution such as https://cloud.google.com/kms/ to help keep secrets safe.
+    private static final String CLOUD_SQL_CONNECTION_NAME = System.getenv("CLOUD_SQL_CONNECTION_NAME");
+    private static final String DB_USER = System.getenv("DB_USER");
+    private static final String DB_PASS = System.getenv("DB_PASS");
+    private static final String DB_NAME = System.getenv("DB_NAME");
+
+    private DataSource createConnectionPool() {
+        // [START cloud_sql_postgres_servlet_create]
+        // The configuration object specifies behaviors for the connection pool.
+        HikariConfig config = new HikariConfig();
+
+        // Configure which instance and what database user to connect with.
+        config.setJdbcUrl(String.format("jdbc:postgresql:///%s", DB_NAME));
+        config.setUsername(DB_USER); // e.g. "root", "postgres"
+        config.setPassword(DB_PASS); // e.g. "my-password"
+
+        // For Java users, the Cloud SQL JDBC Socket Factory can provide authenticated connections.
+        // See https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory for details.
+        config.addDataSourceProperty("socketFactory", "com.google.cloud.sql.postgres.SocketFactory");
+        config.addDataSourceProperty("cloudSqlInstance", CLOUD_SQL_CONNECTION_NAME);
+
+        // ... For future reference, one can also specify additional connection properties here.
+
+        // Initialize the connection pool using the configuration object.
+        DataSource pool = new HikariDataSource(config);
+        // [END cloud_sql_postgres_servlet_create]
+        return pool;
+    }
+
+    private void createUniversityTable(DataSource pool) throws SQLException {
+        // Safely attempts to create the university table schema.
+        try (Connection conn = pool.getConnection()) {
+            String stmt =
+                "CREATE TABLE IF NOT EXISTS university ( "
```

Comment:
Nit: move all the SQL statements into constants towards the top of the file. The table defs are the key piece of info in this file so optimize for a reader seeing those lines as soon as they open the file.